core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GoogleServicesDAO.scala

```diff
@@ -123,6 +123,10 @@ abstract class GoogleServicesDAO(groupsPrefix: String) extends ErrorReportable {
     */
   def listBillingAccountsUsingServiceCredential(implicit executionContext: ExecutionContext): Future[Seq[RawlsBillingAccount]]
 
+  def getProjectBillingAccount(projectName: RawlsBillingProjectName, userInfo: UserInfo)(implicit executionContext: ExecutionContext): Future[Option[String]]
```

Comment:
I've updated the code a little bit to try to clarify this stuff more and referenced a new Jira ticket that I've created.

As discussed over Zoom, it definitely feels like there's going to need to be some refactoring once PPW goes live. There are some things that can be prepped ahead of time but I think there are some re-factorings that are going to be inevitable. 

The thing that concerns me the most is this multi-billing-account state that you've described. If it's the case that the Google projects within a billing project can temporarily map to different billing accounts, then I'm not sure how this is going to work without significant refactoring.

This will require product input but IMO a reasonable way to deal with this inconsistent state would be to prevent users from setting a spend configuration on a billing project that has inconsistent billing accounts across its Google projects. It's already going to be the case that the spend configuration will be cleared when the billing account changes, so there's already some "downtime" on spend reporting anyway in these circumstances.

I offer this solution knowing that I don't have much insight into how frequent a billing project could end up in an inconsistent state and for how long it could last.